Flaky test_hypthesis_sum_agg test

### Describe the bug

The `test_hypothesis_sum_agg` test in `version_store/test_aggregation.py` is flaky.

A fix was made to replace `np.nan` with `np.inf` in the expected pandas df, but in this case the test failed because a `-np.inf` was returned so since `np.inf != -np.inf` the test fails.

Proposed solution would be to leave the `np.nan` in the expected df, but replace the `np.inf` and `-np.inf` occurrences with `np.nan` in the result returned from ArcticDB's query.

See error below:

```
=========================== short test summary info ============================
FAILED tests/unit/arcticdb/version_store/test_aggregation.py::test_hypothesis_sum_agg[EncodingVersion.V2] - AssertionError: DataFrame.iloc[:, 0] (column name="a") are different

DataFrame.iloc[:, 0] (column name="a") values are different (100.0 %)
[index]: [0]
[left]:  [inf]
[right]: [-inf]
At positional index 0, first diff: inf != -inf
Falsifying example: test_hypothesis_sum_agg(
    lmdb_version_store=NativeVersionStore: Library: local.test.822_2023-09-28T10_30_17_503782_v2, Primary Storage: lmdb_storage.,
    df=
          grouping_column              a
        0               0 -4.494233e+307
        1               0 -1.797693e+308
        2               0   1.000000e+00
    ,
)

You can reproduce this example by temporarily adding @reproduce_failure('6.72.4', b'AXicY2ZiYGJgYNz/gAEMoLz/IPDvKEiEEQgZAJggCSw=') as a decorator on your test case
= 1 failed, 6483 passed, 991 skipped, 2 xpassed, 35008 warnings in 3252.45s (0:54:12) =
Error: Process completed with exit code 1.
```

### Steps/Code to Reproduce

Hypothesis test random inputs will cause this.

### Expected Results

No error

### OS, Python Version and ArcticDB Version

macos (conda build) -- GitHub Actions run

See workflow run: https://github.com/man-group/ArcticDB/actions/runs/6336234716/job/17208913448?pr=860

### Backend storage used

_No response_

### Additional Context

_No response_

--- a/python/tests/unit/arcticdb/version_store/test_aggregation.py
+++ b/python/tests/unit/arcticdb/version_store/test_aggregation.py
@@ -63,7 +63,7 @@ def test_hypothesis_mean_agg(lmdb_version_store, df):
     # Older versions of Pandas treat values which exceeds limits as `np.inf` or `-np.inf`.
     # ArcticDB adopted this behaviour.
     #
-    # Yet, wew version of Pandas treats values which exceeds limits as `np.nan` instead.
+    # Yet, new version of Pandas treats values which exceeds limits as `np.nan` instead.
     # To be able to compare the results, we need to replace `np.inf` and `-np.inf` with `np.nan`.
     received_df.replace(-np.inf, np.nan, inplace=True)
     received_df.replace(np.inf, np.nan, inplace=True)

--- a/python/tests/unit/arcticdb/version_store/test_aggregation.py
+++ b/python/tests/unit/arcticdb/version_store/test_aggregation.py
@@ -53,16 +53,22 @@ def test_hypothesis_mean_agg(lmdb_version_store, df):
 
     q = QueryBuilder()
     q = q.groupby("grouping_column").agg({"a": "mean"})
-    expected = df.groupby("grouping_column").agg({"a": "mean"})
-    expected.replace(
-        np.nan, np.inf, inplace=True
-    )  # New version of pandas treats values which exceeds limits as np.nan rather than np.inf, as in old version and arcticdb
+    expected_df = df.groupby("grouping_column").agg({"a": "mean"})
 
     symbol = "mean_agg"
     lib.write(symbol, df)
-    vit = lib.read(symbol, query_builder=q)
-    vit.data.sort_index(inplace=True)
-    assert_frame_equal(expected, vit.data)
+    received_df = lib.read(symbol, query_builder=q).data
+    received_df.sort_index(inplace=True)
+
+    # Older versions of Pandas treat values which exceeds limits as `np.inf` or `-np.inf`.
+    # ArcticDB adopted this behaviour.
+    #
+    # Yet, wew version of Pandas treats values which exceeds limits as `np.nan` instead.
+    # To be able to compare the results, we need to replace `np.inf` and `-np.inf` with `np.nan`.
+    received_df.replace(-np.inf, np.nan, inplace=True)
+    received_df.replace(np.inf, np.nan, inplace=True)
+
+    assert_frame_equal(expected_df, received_df)
 
 
 @use_of_function_scoped_fixtures_in_hypothesis_checked
@@ -82,16 +88,22 @@ def test_hypothesis_sum_agg(lmdb_version_store, df):
 
     q = QueryBuilder()
     q = q.groupby("grouping_column").agg({"a": "sum"})
-    expected = df.groupby("grouping_column").agg({"a": "sum"})
-    expected.replace(
-        np.nan, np.inf, inplace=True
-    )  # New version of pandas treats values which exceeds limits as np.nan rather than np.inf, as in old version and arcticdb
+    expected_df = df.groupby("grouping_column").agg({"a": "sum"})
 
     symbol = "sum_agg"
     lib.write(symbol, df)
-    vit = lib.read(symbol, query_builder=q)
-    vit.data.sort_index(inplace=True)
-    assert_frame_equal(expected, vit.data)
+    received_df = lib.read(symbol, query_builder=q).data
+    received_df.sort_index(inplace=True)
+
+    # Older versions of Pandas treat values which exceeds limits as `np.inf` or `-np.inf`.
+    # ArcticDB adopted this behaviour.
+    #
+    # Yet, wew version of Pandas treats values which exceeds limits as `np.nan` instead.
+    # To be able to compare the results, we need to replace `np.inf` and `-np.inf` with `np.nan`.
+    received_df.replace(-np.inf, np.nan, inplace=True)
+    received_df.replace(np.inf, np.nan, inplace=True)
+
+    assert_frame_equal(expected_df, received_df)
 
 
 @use_of_function_scoped_fixtures_in_hypothesis_checked

--- a/python/tests/unit/arcticdb/version_store/test_aggregation.py
+++ b/python/tests/unit/arcticdb/version_store/test_aggregation.py
@@ -98,7 +98,7 @@ def test_hypothesis_sum_agg(lmdb_version_store, df):
     # Older versions of Pandas treat values which exceeds limits as `np.inf` or `-np.inf`.
     # ArcticDB adopted this behaviour.
     #
-    # Yet, wew version of Pandas treats values which exceeds limits as `np.nan` instead.
+    # Yet, new version of Pandas treats values which exceeds limits as `np.nan` instead.
     # To be able to compare the results, we need to replace `np.inf` and `-np.inf` with `np.nan`.
     received_df.replace(-np.inf, np.nan, inplace=True)
     received_df.replace(np.inf, np.nan, inplace=True)

